### PR TITLE
support osx_arm64 and linux_aarch64

### PR DESCRIPTION
**Pull request by gillins**
Checklist
* [ ] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [ ] Bumped the build number (if the version is unchanged)
* [ ] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [ ] Ensured the license file is being packaged.

Trying to get QGIS working under ARM: https://github.com/conda-forge/qgis-feedstock/pull/289

### CODE REVIEW

**Comment by gillins** [comment]
@conda-forge-admin, please rerender

**Comment by gillins** [comment]
```
$BUILD_PREFIX/bin/../lib/gcc/aarch64-conda-linux-gnu/11.3.0/../../../../aarch64-conda-linux-gnu/bin/ld: $PREFIX/lib/libQt5WebKit.so: undefined reference to `std::condition_variable::wait(std::unique_lock<std::mutex>&)@GLIBCXX_3.4.30'
```

Anyone have any ideas?

**Comment by gillins** [comment]
@conda-forge-admin, please rerender

**Comment by gillins** [comment]
@conda-forge/arm-arch any thoughts on this error? Seems like the wrong `g++` is being picked up on aarch64? Appreciate any pointers, I'm out of ideas.

**Comment by hmaarrfk** [comment]
It is very likely that 5.15 does not support cross compilation on Linux well.

Try building natively on Travis, or through emulation on azure.

File: conda-forge.yml [hunk 1]
```diff
@@ -6,3 +6,9 @@ os_version:
   linux_64: cos7
 conda_build:
   pkg_format: '2'
+build_platform:
+  osx_arm64: osx_64
+  linux_aarch64: linux_64
```
Comment (hmaarrfk) [inline conda-forge.yml]:
```suggestion
  linux_aarch64: linux_aarch64
```

And you can switch between azure or Travis below

**Comment by gillins** [comment]
@conda-forge-admin, please rerender

**Comment by hmaarrfk** [comment]
Is it normal that it seems to be generating bindings for non Webkit stuff too?

**Comment by gillins** [comment]
Hmm I've often wondered that, but it only seems to create a package with the webkit stuff (plus QtDesigner which probably isn't ideal) maybe because the other stuff is already installed by `pyqt`? Happy to take any suggestions on improvements, but I can't see how to restrict what is built. 
Would Travis be faster? Seems the error is now fixed anyway - thanks!

**Comment by hmaarrfk** [comment]
Travis would be faster...... But you have a 55 min limit. 

Qemu is about 5x slower (at best) and typically 6x slower.

It's kinda a wash considering that the limit is 6 hours on azure, and 1 on Travis.

File: conda-forge.yml [hunk 2]
```diff
@@ -6,3 +6,9 @@ os_version:
   linux_64: cos7
 conda_build:
   pkg_format: '2'
+build_platform:
+  osx_arm64: osx_64
+  linux_aarch64: linux_aarch64
+test_on_native_only: true
+provider:
+  linux_aarch64: azure
```
Comment (hmaarrfk) [inline conda-forge.yml]:
```suggestion
  linux_aarch64: travis
```
Lets just try

**Comment by hmaarrfk** [comment]
@conda-forge-admin please rerender

**Comment by hmaarrfk** [comment]
It really does seem to me that it is trying to rebuild all. which is why you might be running out of time.

**Comment by SrNetoChan** [comment]
Well, isn't the source from full pyqt?

If I recall, we tried to include this in PyQt feedstock, but folks don't want it there...

**Comment by SrNetoChan** [comment]
https://github.com/conda-forge/pyqt-feedstock/pull/117

**Comment by SrNetoChan** [comment]
@izahn I know you did lot's of work around this on the pyqt-feedstock. Any ideas?

**Comment by hmaarrfk** [comment]
That conversation indicates that our understanding, that it is recompiling everything is commensurate with the old understanding of others.

I think that it would be appropriate to work to consolidate the two feedstocks

We have made many dependencies optional for pyside2
https://github.com/conda-forge/pyside2-feedstock/blob/main/recipe/meta.yaml#L31

See how the tests down the line are adapted.

File: recipe/build.sh [hunk 3]
```diff
@@ -56,5 +57,7 @@ if [[ "${CONDA_BUILD_CROSS_COMPILATION:-}" == "1" ]]; then
   mv Makefile.temp Makefile
 fi
 
-CPATH=$PREFIX/include make -j$CPU_COUNT
+# just build the parts we need
+CPATH=$PREFIX/include make -j$CPU_COUNT sub-QtWebKit
+CPATH=$PREFIX/include make -j$CPU_COUNT sub-QtWebKitWidgets
```
Comment (hmaarrfk) [inline recipe/build.sh:61]: interesting

Comment (gillins) [reply recipe/build.sh:61]: Yeah it's a bit messy but hopefully I can get it to work. I guess if we can get this going here without rocking the the `pyqt` boat that would be best, we understand that `qtwebkit` is seriously un-cool these days but QGIS still needs it (see discussion at https://github.com/conda-forge/qgis-feedstock/pull/242).

Comment (hmaarrfk) [reply recipe/build.sh:61]: Yeah, "high touch" recipes are hard to break into. This is great actually, I think a step in the right "direction".

**Comment by hmaarrfk** [comment]
With regards to my old comment. I think the approach here strikes the right balance, between some optimization that is specific to conda-forge's unique constraints (i.e. build time) and having more granular control over dependencies.

**Comment by gillins** [comment]
I agree, thanks for your help and nudging me to have a closer look at how things were working 'under the hood'. The `sip` stuff is a mystery to me....

**Comment by SrNetoChan** [comment]
Uhmm, seems that something strange happened in the final build. Travis failed to pass. Since this is much faster now, maybe we can just go back to CI for everything?

**Comment by gillins** [comment]
```
No output has been received in the last 10m0s, this potentially indicates a stalled build or something wrong with the build itself.
```
Weird, might have to go back to Azure.

**Pull request merged**